packages/building-rollup/modern-and-legacy-config.js

```diff
@@ -107,6 +109,23 @@ function createConfig(_options, legacy) {
 
       // hash
       entrypointHashmanifest(),
+
+      production &&
```

Comment:
As discussed I think it's clearer to add a boolean option to prevent adding the workbox plugin at all. This way people either go with our defaults, or add their own configuration. The 'extra' config option is just confusing, I think. We should do the same for the other plugins.

Reply:
So something like:

```js
export default createBasicConfig({ 
  input: './index.html',
  useWorkbox: true
  // optional (otherwise use default settings):
  // workboxConfig: require('./workbox-config.js),
});

```

Reply:
Only booleans for the different plugins (they all default to true):

```js
export default createBasicConfig({ 
  input: './index.html',
  plugins: {
    workbox: false,
    babel: false,
    indexHtml: false,
  }
});
```

And then if you want to customize workbox, you do:

```js
const basicConfig = createBasicConfig({ 
  input: './index.html',
  plugins: {
    workbox: false,
    babel: false,
    indexHtml: false,
  }
});

export default merge(basicConfig, {
  plugins: [
    workbox(/*  own config */)
  ]
});
```

Reply:
Where is  `merge()` from?

Reply:
Deepmerge, we use it all over the place.

Reply:
By that I mean it should be easy to find an example somewhere

Reply:
Hahah I got it, no worries 😛